FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/mbedtls_freertos.c

```diff
@@ -0,0 +1,541 @@
+ *
+ * @param[in] pSslContext The SSL context to free.
+ */
+static void sslContextFree( SSLContext_t * pSslContext );
+
+/**
+ * @brief Set up TLS on a TCP connection.
+ *
+ * @param[in] pNetworkContext Network context.
+ * @param[in] pHostName Remote host name, used for server name indication.
+ * @param[in] pNetworkCredentials TLS setup parameters.
+ *
+ * @return #MBEDTLS_SUCCESS, #MBEDTLS_INSUFFICIENT_MEMORY, #MBEDTLS_INVALID_CREDENTIALS,
+ * #MBEDTLS_HANDSHAKE_FAILED, or #MBEDTLS_API_ERROR.
+ */
+static MbedtlsStatus_t tlsSetup( NetworkContext_t * pNetworkContext,
+                                 const char * pHostName,
+                                 const NetworkCredentials_t * pNetworkCredentials );
+
+/**
+ * @brief Initialize mbedTLS.
+ *
+ * @return #MBEDTLS_SUCCESS, or #MBEDTLS_API_ERROR.
+ */
+static MbedtlsStatus_t initMbedtls( void );
+
+/*-----------------------------------------------------------*/
+
+static void sslContextInit( SSLContext_t * pSslContext )
+{
+    mbedtls_ssl_config_init( &( pSslContext->config ) );
+    mbedtls_x509_crt_init( &( pSslContext->rootCa ) );
+    mbedtls_pk_init( &( pSslContext->privKey ) );
+    mbedtls_x509_crt_init( &( pSslContext->clientCert ) );
+    mbedtls_ssl_init( &( pSslContext->context ) );
+}
+/*-----------------------------------------------------------*/
+
+static void sslContextFree( SSLContext_t * pSslContext )
+{
+    mbedtls_ssl_free( &( pSslContext->context ) );
+    mbedtls_x509_crt_free( &( pSslContext->rootCa ) );
+    mbedtls_x509_crt_free( &( pSslContext->clientCert ) );
+    mbedtls_pk_free( &( pSslContext->privKey ) );
+    mbedtls_ssl_config_free( &( pSslContext->config ) );
+}
+
+/*-----------------------------------------------------------*/
+
+static MbedtlsStatus_t tlsSetup( NetworkContext_t * pNetworkContext,
+                                 const char * pHostName,
+                                 const NetworkCredentials_t * pNetworkCredentials )
+{
+    MbedtlsStatus_t returnStatus = MBEDTLS_SUCCESS;
+    int mbedtlsError = 0;
+
+    /* Initialize the mbed TLS context structures. */
+    sslContextInit( &( pNetworkContext->sslContext ) );
+
+    mbedtlsError = mbedtls_ssl_config_defaults( &( pNetworkContext->sslContext.config ),
+                                                MBEDTLS_SSL_IS_CLIENT,
+                                                MBEDTLS_SSL_TRANSPORT_STREAM,
+                                                MBEDTLS_SSL_PRESET_DEFAULT );
+
+    if( mbedtlsError != 0 )
+    {
+        LogError( ( "Failed to set default SSL configuration, error %d.", mbedtlsError ) );
+
+        /* Per mbed TLS docs, mbedtls_ssl_config_defaults only fails on memory allocation. */
+        returnStatus = MBEDTLS_INSUFFICIENT_MEMORY;
+    }
+
+    if( returnStatus == MBEDTLS_SUCCESS )
+    {
+        /* Set up the certificate security profile, starting from the default value. */
+        pNetworkContext->sslContext.certProfile = mbedtls_x509_crt_profile_default;
```

Comment:
As the `certProfile` member is only used in this function for setting the configuration in ` pNetworkContext->sslContext.config`, can this member be removed from `SSLContext_t` type?

Reply:
No, the memory needs to be kept as long as config as active as I understood. It needs to be part of the SSLContext_t